chemistry/structure.py

```diff
@@ -919,11 +919,19 @@ def update_dihedral_exclusions(self):
         turns it to `True` if the two end atoms are in the bond or angle
         partners arrays
         """
+        list14 = []
```

Comment:
A `set` is basically just the key-half of a `dict`, and `dict` has a O(1) lookup (which is why all of Python's innards, like attribute lookup and such) is done using a `dict`.

There are some caveats though -- `set`s are unordered (and I don't know about an `OrderedDict` analogy), and you can only add hashable objects to it (like tuples and other immutable types).  One of the reasons `Atom` does not have a `__eq__` method comparing all of its attributes is because that ruins hashability in Python 3 (and I didn't feel like implementing my own `__hash__` function to circumvent it).